File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Resolutions/ResolutionRole.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Commons;

namespace VecompSoftware.DocSuiteWeb.Entity.Resolutions
{
    public class ResolutionRole : DSWBaseEntity
    {
        #region [ Constructor ]
        public ResolutionRole() : this(Guid.NewGuid()) { }
        public ResolutionRole(Guid uniqueId)
            : base(uniqueId)
        {
        }
        #endregion

        #region [ Properties ]

        public int IdResolutionRoleType { get; set; }
        #endregion

        #region [ Navigation Properties ]

        public virtual Resolution Resolution { get; set; }

        public virtual Role Role { get; set; }

        #endregion


    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Resolution/ResolutionRole.vb
<Serializable()> _
Public Class ResolutionRole
    Inherits DomainObject(Of ResolutionRoleCompositeKey)
    Implements IAuditable

#Region "Properties"
    Public Overridable Property RegistrationUser As String Implements IAuditable.RegistrationUser
    Public Overridable Property RegistrationDate As DateTimeOffset Implements IAuditable.RegistrationDate
    Public Overridable Property LastChangedUser As String Implements IAuditable.LastChangedUser
    Public Overridable Property LastChangedDate As DateTimeOffset? Implements IAuditable.LastChangedDate

    Public Overridable Property Resolution As Resolution
    Public Overridable Property Role As Role
    Public Overridable Property ResolutionRoleType As ResolutionRoleType
    Public Overridable Property UniqueIdResolution As Guid
#End Region

#Region "Ctor/init"
    Public Sub New()
        Id = New ResolutionRoleCompositeKey
        UniqueId = Guid.NewGuid()
    End Sub
#End Region

End Class

